src/nmodl/solve.cpp

```diff
@@ -125,21 +132,20 @@ void solvhandler() {
         qsol = ITM(lq);
         lq = lq->next;
         method = SYM(lq);
-        cvodemethod_ = 0;
-        if (method && strcmp(method->name, "after_cvode") == 0) {
-            method = (Symbol*) 0;
-            lq->element.sym = (Symbol*) 0;
-            cvodemethod_ = 1;
-        }
-        if (method && strcmp(method->name, "cvode_t") == 0) {
-            method = (Symbol*) 0;
-            lq->element.sym = (Symbol*) 0;
-            cvodemethod_ = 2;
-        }
-        if (method && strcmp(method->name, "cvode_v") == 0) {
-            method = (Symbol*) 0;
-            lq->element.sym = (Symbol*) 0;
-            cvodemethod_ = 3;
+        cvodemethod_ = CVodeMethod::nomethod;
+        const std::vector<std::pair<std::string, CVodeMethod>> cvode_methods = {
+            {"after_cvode", CVodeMethod::after_cvode},
+            {"cvode_t", CVodeMethod::cvode_t},
+            {"cvode_t_v", CVodeMethod::cvode_t_v}};
+
+        const auto& method_match =
+            std::find_if(cvode_methods.begin(), cvode_methods.end(), [&method](const auto& item) {
+                return method && method->name == item.first;
+            });
+        if (method && method_match != cvode_methods.end()) {
+            cvodemethod_ = method_match->second;
```

Comment:
So... this is basically like a `map` lookup? (your code finds the match before verifying that method is valid, too)
```suggestion
        const std::map<std::string, CVodeMethod> cvode_methods = {
            {"after_cvode", CVodeMethod::after_cvode},
            {"cvode_t", CVodeMethod::cvode_t},
            {"cvode_t_v", CVodeMethod::cvode_t_v}};

        decltype(cvode_methods)::const_iterator m;
        if (method && (m = cvode_methods.find(method)) != cvode_methods.end()) {
                    cvodemethod_ = m->second;
```